doc/configurable.md

```diff
@@ -0,0 +1,206 @@
+        x = x.view(-1, 16 * 5 * 5)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+config = dict(
+    factory=Net,
+    conv1=dict(
+        factory=nn.Conv2d,
+        in_channels=3,
+        out_channels=6,
+        kernel_size=5
+    ),
+    pool=dict(
+        factory=nn.MaxPool2d,
+        kernel_size=2,
+        stride=2
+    ),
+    conv2=dict(
+        factory=nn.Conv2d,
+        in_channels=6,
+        out_channels=16,
+        kernel_size=5
+    ),
+    fc1=dict(
+        factory=nn.Linear,
+        in_features=16*5*5,
+        out_features=120
+    ),
+    fc2=dict(
+        factory=nn.Linear,
+        in_features=120,
+        out_features=84
+    ),
+    fc3=dict(
+        factory=nn.Linear,
+        in_features=84,
+        out_features=10
+    ),
+)
+net = pt.Configurable.from_config(config)
+```
+This initialization looks more complicated than the two possiblities above,
+but it has major upsites.
+The main benefit is that we can now easily adjust the parameters from outside
+of the model.
+Another benefit is, that this initialization allows to easily save the used
+configuration for example by writting the config dictionary to a json file.
+However, the current config does not include all parameters used in the model
+since there are some default values in the modules which are not specified in
+the config dict.
+If Net is a sub-class of Configurable (for example by replacing nn.Module with
+padertorch.Module) we can get a config with all parameters in the model by
+calling ```Net.get_config(config)```:
+```python
+import torch.nn as nn
+import torch.nn.functional as F
+import padertorch as pt
+
+class Net(pt.Module):
+    def __init__(self, conv1, pool, conv2, fc1, fc2, fc3):
+        super().__init__()
+        self.conv1 = conv1
+        self.pool = pool
+        self.conv2 = conv2
+        self.fc1 = fc1
+        self.fc2 = fc2
+        self.fc3 = fc3
+
+    def forward(self, x):
+        x = self.pool(F.relu(self.conv1(x)))
+        x = self.pool(F.relu(self.conv2(x)))
+        x = x.view(-1, 16 * 5 * 5)
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+        return x
+
+config = dict(
+    factory=Net,
+    conv1=dict(
+        factory=nn.Conv2d,
+        in_channels=3,
+        out_channels=6,
+        kernel_size=5
+    ),
+    pool=dict(
+        factory=nn.MaxPool2d,
+        kernel_size=2,
+        stride=2
+    ),
+    conv2=dict(
+        factory=nn.Conv2d,
+        in_channels=6,
+        out_channels=16,
+        kernel_size=5
+    ),
+    fc1=dict(
+        factory=nn.Linear,
+        in_features=16*5*5,
+        out_features=120
+    ),
+    fc2=dict(
+        factory=nn.Linear,
+        in_features=120,
+        out_features=84
+    ),
+    fc3=dict(
+        factory=nn.Linear,
+        in_features=84,
+        out_features=10
+    ),
+)
+full_config = Net.get_config(config)
```

Comment:
Could you add `pt.Configurable.get_config(config)` as alternative, when `Net` does not inherent from `pt.Module`?

In general `Configurable` is some kind of "duck" type.
As along as the `__init__` (or function) has no positional only argument (in discussion #61) and all arguments are  `Configurable` the class (or function) is `Configurable`.